kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/hspa/HspaModem.java

```diff
@@ -40,703 +40,725 @@
+				closeSerialPort(commAtConnection);
+				throw new KuraException(KuraErrorCode.NOT_CONNECTED, MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG);
+			}
+			try {
+				reply = commAtConnection.sendCommand(HspaModemAtCommands.getMobileStationClass.getCommand().getBytes(),
+						1000, 100);
+			} catch (IOException e) {
+				closeSerialPort(commAtConnection);
+				throw new KuraException(KuraErrorCode.CONNECTION_FAILED, e);
+			}
+			closeSerialPort(commAtConnection);
+			if (reply != null) {
+				String sCgclass = this.getResponseString(reply);
+				if (sCgclass.startsWith("+CGCLASS:")) {
+					sCgclass = sCgclass.substring("+CGCLASS:".length()).trim();
+					if ("\"A\"".equals(sCgclass)) {
+						serviceType = "UMTS";
+					} else if ("\"B\"".equals(sCgclass)) {
+						serviceType = "GSM/GPRS";
+					} else if ("\"CG\"".equals(sCgclass)) {
+						serviceType = "GPRS";
+					} else if ("\"CC\"".equals(sCgclass)) {
+						serviceType = "GSM";
+					}
+				}
+			}
+		}
+
+		return serviceType;
+	}
+
+	@Override
+	public List<ModemTechnologyType> getTechnologyTypes() throws KuraException {
+
+		List<ModemTechnologyType> modemTechnologyTypes;
+		ModemDevice dev = getModemDevice();
+		if (dev == null) {
+			throw new KuraException(KuraErrorCode.UNAVAILABLE_DEVICE, "No modem device");
+		}
+		if (dev instanceof UsbModemDevice) {
+			SupportedUsbModemInfo usbModemInfo = SupportedUsbModemsInfo.getModem((UsbModemDevice) dev);
+			if (usbModemInfo != null) {
+				modemTechnologyTypes = usbModemInfo.getTechnologyTypes();
+			} else {
+				throw new KuraException(KuraErrorCode.UNAVAILABLE_DEVICE, "No usbModemInfo available");
+			}
+		} else {
+			throw new KuraException(KuraErrorCode.UNAVAILABLE_DEVICE, "Unsupported modem device");
+		}
+		return modemTechnologyTypes;
+	}
+
+	@Override
+	public boolean isSimCardReady() throws KuraException {
+		boolean status = false;
+		synchronized (this.atLock) {
+			logger.debug("sendCommand isSimCardReady :: {}", HspaModemAtCommands.getSimPinStatus.getCommand());
+			byte[] reply;
+			CommConnection commAtConnection = openSerialPort(getAtPort());
+			if (!isAtReachable(commAtConnection)) {
+				closeSerialPort(commAtConnection);
+				throw new KuraException(KuraErrorCode.NOT_CONNECTED, MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG);
+			}
+			try {
+				reply = commAtConnection.sendCommand(HspaModemAtCommands.getSimPinStatus.getCommand().getBytes(), 1000,
+						100);
+			} catch (IOException e) {
+				closeSerialPort(commAtConnection);
+				throw new KuraException(KuraErrorCode.CONNECTION_FAILED, e);
+			}
+			closeSerialPort(commAtConnection);
+			if (reply != null) {
+				String sReply = new String(reply);
+				if (sReply.contains("OK")) {
+					status = true;
+				}
+			}
+		}
+		return status;
+	}
+
+	private String formGetPdpContextAtCommand() {
+		StringBuilder sb = new StringBuilder(HspaModemAtCommands.pdpContext.getCommand());
+		sb.append("?\r\n");
+		return sb.toString();
+	}
+
+	@Override
+	public boolean isModemLTE() {
```

Comment:
I'm not sure about this API. Isn't better to have a getModemType that returns a value in an enum? Maybe it is more flexible and future proof

Reply:
isModemLTE is not the right denomination, maybe hasDiversityAntenna would be more meaningful?
I don't see the benefit of returning an enum value, in any case if it's true it will have to implement enable/disable diversity?

Reply:
Could be